src/XML/src/FMI3/fmi3_xml_vendor_annotations.c

```diff
@@ -78,30 +78,30 @@ int fmi3_xml_handle_VariableTool(fmi3_xml_parser_context_t *context, const char*
 
 int fmi3_xml_handle_Tool(fmi3_xml_parser_context_t *context, const char* data) {
     if(!data) {
-            size_t len;
-            fmi3_xml_model_description_t* md = context->modelDescription;
-            jm_vector(char)* bufName = fmi3_xml_reserve_parse_buffer(context,1,100);
-            jm_string *pvendor;
-            char* vendor = 0;
-
-            if(!bufName) return -1;
-            /* <xs:attribute name="name" type="xs:normalizedString" use="required"> */
-            if( fmi3_xml_parse_attr_as_string(context, fmi3_xml_elmID_Tool, fmi_attr_id_name, 1, bufName))
-                return -1;
-            pvendor = jm_vector_push_back(jm_string)(&md->vendorList, vendor);
-            len = jm_vector_get_size(char)(bufName);
-            if(pvendor )
-                *pvendor = vendor = (char*)(context->callbacks->malloc(len + 1));
-            if(!pvendor || !vendor) {
-                fmi3_xml_parse_fatal(context, "Could not allocate memory");
-                return -1;
-            }
-            memcpy(vendor, jm_vector_get_itemp(char)(bufName,0), len);
-            vendor[len] = 0;
-
-            context->anyToolName = vendor;
-            context->anyParent = 0;
-            context->useAnyHandleFlg = 1;
+        size_t len;
+        fmi3_xml_model_description_t* md = context->modelDescription;
+        jm_vector(char)* bufName = fmi3_xml_reserve_parse_buffer(context,1,100);
+        jm_string *pvendor;
+        char* vendor = 0;
+
+        if(!bufName) return -1;
+        /* <xs:attribute name="name" type="xs:normalizedString" use="required"> */
+        if( fmi3_xml_parse_attr_as_string(context, fmi3_xml_elmID_Tool, fmi_attr_id_name, 1, bufName))
+            return -1;
+        pvendor = jm_vector_push_back(jm_string)(&md->vendorList, vendor);
+        len = jm_vector_get_size(char)(bufName);
+        if(pvendor )
```

Comment:
```suggestion
        if (pvendor)
```
maybe we should add a `{ }` as well